### ISSUE TITLE
Help with API - setMyCommands

### ISSUE THREAD

**Issue by kdabwl** [vendelieu/telegram-bot#1]
Hello,
I've already tried .js implementations and nodejs http request to get some positive result when using setMyCommands. The request/response says O.K. but nothing was changed (tried several settings with @telegram‧botFather on my bot, as well).

If telegram-bot can show that setMyCommands works, I would have to switch to Kotlin.

P.S. clever, the interface BotWaitingInput

**Comment by vendelieu** [comment]
Hello, I do not often use the full range of Telegram API, so the interface of some commands may not be entirely convenient, but in any case, having checked now everything is quite working.


```
setMyCommands(
            null, null,
            BotCommand("/test", "test command"), BotCommand("/test2", "test command")
        )
```
 
```
Response: {
  "ok" : true,
  "result" : true
}
```

![изображение](https://user-images.githubusercontent.com/3987067/172227112-748a17e5-f651-4c5c-8c49-352f4db35fc2.png)


The first two parameters are `scope` and `language_code`, which are optional, so that's why there is null.
I think in a future release we should change the interface of the command.

**Comment by vendelieu** [comment]
I also checked with scope

works very well:
```
deleteMyCommands().send(bot)
setMyCommands(
        languageCode = null,
        scope = BotCommandScope.AllGroupChats,
        BotCommand("/test1", "test command")
).send(bot)
```
Bot Private chat after request:
![изображение](https://user-images.githubusercontent.com/3987067/172374465-7071fe1d-4e44-4e9c-af42-69caf69aafbd.png)

In group:
![изображение](https://user-images.githubusercontent.com/3987067/172374417-357fb4f8-dc29-4ebb-9262-c7712b05ff70.png)

**Comment by vendelieu** [comment]
While checking commands, I found a small bug on polymorphic deserialization of sealed classes, a fix for which I already committed in `v1.1.2`

In the process of manual testing, because it was implemented all the Telegram API was difficult to find all the possible bugs, but in the near future I plan to cover everything with the program tests. 

If there will be any feedback or suggestions on the interface or anything else, I will be very happy to listen.

**Comment by kdabwl** [comment]
Commandy and Menu button are only a small aspect for our budget and client - rather a look and feel, compared to the main work processing. I cannot promise more than putting prototype up for discussion here.

**Comment by vendelieu** [comment]
Yes, I will wait with interest for your questions and if there are suggestions.

**Comment by kdabwl** [comment]
Hello, I mentioned the requirement for [setChatMenuButton](https://core.telegram.org/bots/api#setchatmenubutton) above; can we implement that ourselves by using existing functionality of telegram-bot, unchanged?
Depending on words (or other condition) sent in a message to the bot, it should set the menu button to {text: 'any', url: 'https:'+'//something'}, but only for use in chat with the sender (I think that is private chat in Telegram terminology).

**Comment by vendelieu** [comment]
Yeah, there's same function [setChatMenuButton()](https://vendelieu.github.io/telegram-bot/-telegram%20-bot/com.github.vendelieu.tgbot.api.chat/set-chat-menu-button.html)

```
setChatMenuButton(
  MenuButton.WebApps("test", WebAppInfo("webappUrl"))
).send(1234, bot)
```

**Comment by vendelieu** [comment]
recently released version `1.2.2`, where you can use a more user-friendly interface of `setMyCommands()` action
```
setMyCommands {
    botCommand("/test", "description, of command")
    botCommand("/test2", "another command")
}
```
Also, there are improvements in interface of some other functions.
plus, you can introduce your "magic object" via interface
